module-1/src/Components/Genres/Genres.tsx

```diff
@@ -1,29 +1,55 @@
 import { useState } from "react";
 
-interface IGenresProps{
-    genres:string[],
-    selected:string,
-    onSelect: (val:string)=>void
+interface IGenresProps {
+  /**
+   * This is the entire list of Genres availables in the Component
+   */
+  genres: string[];
+
+  /**
+   * This is the single Genre that will be selected by Default when the component renders. Is Required.
+   */
+  selected: string;
+
+  /**
+   * This is the callback function that will be executed when a Genre is selected. Is Required.
+   */
+  onSelect: (val: string) => void;
 }
 
-const Genres:React.FC<IGenresProps> = ({genres,selected,onSelect}) => {
-    return (
-        <div className='border p-3 m-3'>
-          <h3 className='text-lg m-4 font-bold'>Genres Component</h3>
-          {genres.map((genre, index) => (
-            <button key={index} onClick={() => onSelect(genre)} 
-            className={`inline-block bg-${genre === selected ? 'green' : 'white'} 
+export const Genres = ({ genres, selected, onSelect }: IGenresProps) => {
+  const [selectedGenre, setSelectedGenre] = useState(selected);
+  const handleGenreSelection = (genre: string) => {
+    setSelectedGenre(genre);
+    onSelect(genre);
+  };
+  return (
+    <div className="border p-3 m-3">
+      <h3 className="text-lg m-4 font-bold">Genres Component</h3>
+      {genres.map((genre, index) => (
+        <button
+          key={index}
```

Comment:
React team recommends against using `index` as keys, more context here https://react.dev/learn/rendering-lists#rules-of-keys

Reply:
fixed